### ISSUE TITLE
Review all xfail and skip markers in tests

### ISSUE THREAD

**Issue by konradkonrad** [raiden-network/raiden#397]
PR #391 unearthed some missed bugs and regressions that were missed due to the skipped test on CI. We should take some time to ensure all `xfail`s are reasonable and `skip`s are really justified. See also #389 

## xfail and skip tests

- [ ] unit
  - [x] test_lock_timeout_lower_than_previous_channel_settlement_period: xfail, not implemented
  - [x] test_do_not_withdraw_an_almost_expiring_lock_if_a_payment_didnt_occur: xfail, not implemented (see #382)
  - [ ] test_insufficient_funds: xfail
  - [ ] test_transfer_succesful_after_secret_learned: xfail, not implemented (see #522)
- [x] smart_contracts
  - [x] test_close_valid_tranfer_different_token: xpass, depending on #292
- [x] property
- [x] long_running
  - [x] test_participant_selection: xfail, see #691
- [ ] integration
  - [ ] test_secret_revealed: xfail
  - [x] test_blockchain: skip on travis, see #319
  - [ ] test_start_end_attack: xfail, test incomplete
- [x] api

**Comment by LefterisJP** [comment]
Also important. For each skipped or xfailed test have a separate issue.

**Comment by hackaugusto** [comment]
echo_node tests were marked as skip before merging the channel state PR

**Comment by LefterisJP** [comment]
Closing this issue as it serves no purpose any longer. We have, and should keep creating if needed, individual issues for any skipped or xfailed tests.
